vector/src/main/java/im/vector/app/features/settings/devices/v2/othersessions/OtherSessionsFragment.kt

```diff
@@ -0,0 +1,164 @@
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.core.resources.ColorProvider
+import im.vector.app.databinding.FragmentOtherSessionsBinding
+import im.vector.app.features.settings.devices.v2.DeviceFullInfo
+import im.vector.app.features.settings.devices.v2.DevicesAction
+import im.vector.app.features.settings.devices.v2.DevicesViewModel
+import im.vector.app.features.settings.devices.v2.VectorSettingsDevicesViewNavigator
+import im.vector.app.features.settings.devices.v2.filter.DeviceManagerFilterBottomSheet
+import im.vector.app.features.settings.devices.v2.filter.DeviceManagerFilterType
+import im.vector.app.features.settings.devices.v2.list.OtherSessionsView
+import im.vector.app.features.settings.devices.v2.list.SESSION_IS_MARKED_AS_INACTIVE_AFTER_DAYS
+import im.vector.app.features.themes.ThemeUtils
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class OtherSessionsFragment :
+        VectorBaseFragment<FragmentOtherSessionsBinding>(),
+        VectorBaseBottomSheetDialogFragment.ResultListener,
+        OtherSessionsView.Callback {
+
+    private val viewModel: DevicesViewModel by fragmentViewModel()
+    @Inject lateinit var colorProvider: ColorProvider
+    @Inject lateinit var viewNavigator: VectorSettingsDevicesViewNavigator
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentOtherSessionsBinding {
+        return FragmentOtherSessionsBinding.inflate(layoutInflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupToolbar(views.otherSessionsToolbar).allowBack()
+        initFilterView()
+    }
+
+    private fun initFilterView() {
+        views.otherSessionsFilterFrameLayout.debouncedClicks {
+            withState(viewModel) { state ->
+                DeviceManagerFilterBottomSheet
+                        .newInstance(state.currentFilter, this)
+                        .show(requireActivity().supportFragmentManager, "SHOW_DEVICE_MANAGER_FILTER_BOTTOM_SHEET")
+            }
+        }
+
+        views.otherSessionsClearFilterButton.debouncedClicks {
+            viewModel.handle(DevicesAction.FilterDevices(DeviceManagerFilterType.ALL_SESSIONS))
+        }
+
+        views.deviceListOtherSessions.callback = this
+    }
+
+    override fun onBottomSheetResult(resultCode: Int, data: Any?) {
+        if (resultCode == RESULT_OK && data != null && data is DeviceManagerFilterType) {
+            viewModel.handle(DevicesAction.FilterDevices(data))
+        }
+    }
+
+    override fun invalidate() = withState(viewModel) { state ->
+        if (state.devices is Success) {
+            with(state) {
+                val devices = state.devices()
+                        ?.filter { it.deviceInfo.deviceId != state.currentSessionCrossSigningInfo.deviceId }
```

Comment:
I guess filtering out the current session should be some way into a `UseCase` inside the `ViewModel` so that we are able to test it. What do you think?

Reply:
Moved to the new use case.